Updated Kafka Scalar documentation to include new authmode for Event Hubs (#78)
kedacore/keda-docs: content/scalers/apache-kafka-topic.md

background = "light"
+++

> Scale applications based on Apache Kafka Topic. This can also be used to scale applications based on Azure Event Hubs w/Kafka enabled.

Reviewer: To be honest I don't think that this kind of "specific" use case should be mentioned. Who is using AEH with Kafka head knows that can use this trigger because ... it's about Kafka topic.

Reviewer 2: That's a fair point, what about this instead @ppatierno?
```suggestion
Scale applications based on Apache Kafka Topic or other services that support Kafka protocol.
```
Or:
```suggestion
Scale applications based on Apache Kafka Topic or other services that support Kafka protocol such as Azure Event Hubs.
```

Author: Thanks for the feedback!
I definitely like the 2nd recommendation made by @tomkerkhove

Reviewer: I am more for the first one, because if at some point another service will come supporting the Kafka protocol, then we should mention it here as well. For this reason, I would be more generic.
Then, of course, having a dedicated example of using Azure EH (as I mentioned in another comment).

 You can use `TriggerAuthentication` CRD to configure the authenticate by providing authMode, username, password. If your kafka cluster does not have sasl authentication turned on, you will not need to pay attention to it.

>  The sasl_ssl_plain authMode can be used when using the Kafka head on Azure Event Hubs to act as trigger.

Reviewer: Same as above, who uses AEH knows that it supports SASL_SSL_PLAIN so the trigger has to be configured in that way.

Reviewer 2: I think this is a fair point in the documentation, what is your concern with this? We provide several means to authenticate and here we just point out that this works with Event Hubs, don't see an issue with that.

Author: @ppatierno, actually anyone who has played with producers and consumers to AEH w/Kafka endpoint would know that this is the way to authenticate with it. I feel the mention here does more good than harm :)

Reviewer: I see it the other way around. :-) The point I am making here is that the user who uses EH and is interested in using it via Kafka's head should read the Microsoft official doc to check what kind of authentication it supports. At that point, if he wants to use KEDA then he should already know to set `sasl_ssl_plain` as auth mode so it's not worth mentioning EH here.

Author: I've modified the doc to read `The sasl_ssl_plain authMode can also be used when using the Kafka head on Azure Event Hubs to act as trigger.`

Reviewer: Which again should be something that people using Azure EH with Kafka should already know :-)
As the same point I tried to make before: imagine that the project Apache Pulsar will add a Kafka protocol head, at this point we should update the doc saying " ... when using the Kafka head on Azure Event Hubs or Apache Pulsar to act as trigger". It should happen again and again for other new services with same support. I would be more generic not mentioning specific cases but leaving them as dedicated examples.

Author: Have made the requested changes

      lagThreshold: "50"
    authenticationRef:
      name: keda-trigger-auth-kafka-credential

> ```

Reviewer: not clear what's the difference here may be git is going crazy :-)

Reviewer 2: Think it's a diff thing for sure!

      lagThreshold: "50"
    authenticationRef:
      name: keda-trigger-auth-kafka-credential
```

> Your kafka head on Azure Event Hubs - turn on sasl ssl plain auth

Reviewer: It should be a more generic example about SASL_SSL_PLAIN not specific to AEH.
This new added mechanism works with any Kafka "installation" not just AEH. From this doc, it could be misled imho.

Reviewer 2: Agreed, I'd delete this line here or new section as mentioned below.

Author: point taken! Thanks for the feedback :)

> For more details on how to get the Event Hub Namespace and Event Hub ConnectionString, refer to documentations [here](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quickstart-kafka-enabled-event-hubs).

Reviewer: I don't think that a "specific" example like this should be part of the generic Kafka topic trigger documentation. Maybe we need more a section for examples.

Reviewer 2: Event Hubs speaks Kafka so it's a fair point to note in the scaler docs here, but I'd suggest to introduce a "Using Kafka with other implementations" section and move it under there, ok for you?

Author: I guess in this case I will go with just providing a sample in the Samples documentation instead. Would that be ok?

Reviewer: Yes, I agree with having a dedicated sample for it.
